File: ceemsp-frontend/src/app/_models/EmpresaDomicilio.ts
import { Deserializable } from "./Deserializable";
import Estado from "./Estado";
import Municipio from "./Municipio";
import Localidad from "./Localidad";
import Colonia from "./Colonia";
import Calle from "./Calle";

export default class EmpresaDomicilio implements Deserializable {
  id: number;
  uuid: string;
  nombre: string;
  domicilio1: string;
  numeroExterior: string;
  numeroInterior: string;
  domicilio2: string;
  domicilio3: string;
  domicilio4: string;
  localidad: string;
  estado: string;
  pais: string;
  codigoPostal: string;
  matriz: string;
  telefonoFijo: string;
  telefonoMovil: string;
  latitud: string;
  longitud: string;
  estadoCatalogo: Estado;
  municipioCatalogo: Municipio;
  localidadCatalogo: Localidad;
  coloniaCatalogo: Colonia;
  calleCatalogo: Calle;

  motivoBaja: string;
  observacionesBaja: string;
  documentoFundatorioBaja: string;
  fechaBaja: string;

  deserialize(input: any): this {
    Object.assign(this, input);
    return this;
  }

  static obtenerColumnasPorDefault() {
    return [
      {headerName: 'ID', field: 'uuid', sortable: true, filter: true},
      {headerName: 'Nombre', field: 'nombre', sortable: true, filter: true },
      {headerName: 'Domicilio', field: 'domicilio1', sortable: true, filter: true},
      {headerName: 'Colonia', field: 'domicilio2', sortable: true, filter: true},
      {headerName: 'Estado', field: 'estado', sortable: true, filter: true},
      {headerName: 'C.P.', field: 'codigoPostal', sortable: true, filter: true}
    ];
  }

  static obtenerTodasLasColumnas() {
    return [
      {headerName: 'ID', field: 'uuid', sortable: true, filter: true},
      {headerName: 'Nombre', field: 'nombre', sortable: true, filter: true },
      {headerName: 'Domicilio', field: 'domicilio1', sortable: true, filter: true},
      {headerName: 'N. Ext', field: 'numeroExterior', sortable: true, filter: true},
      {headerName: 'N. Int', field: 'numeroInterior', sortable: true, filter: true},
      {headerName: 'Colonia', field: 'domicilio2', sortable: true, filter: true},
      {headerName: 'Municipio', field: 'domicilio3', sortable: true, filter: true},
      {headerName: 'Referencia', field: 'domicilio4', sortable: true, filter: true},
      {headerName: 'Estado', field: 'estado', sortable: true, filter: true},
      {headerName: 'Pais', field: 'pais', sortable: true, filter: true},
      {headerName: 'C.P.', field: 'codigoPostal', sortable: true, filter: true},
      {headerName: 'Tel. Fijo', field: 'telefonoFijo', sortable: true, filter: true},
      {headerName: 'Tel. Movil', field: 'telefonoMovil', sortable: true, filter: true}
    ]
  }
}

File: database/src/main/java/com/pelisat/cesp/ceemsp/database/model/EmpresaDomicilio.java
package com.pelisat.cesp.ceemsp.database.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Getter
@Setter
@Entity
@Table(name = "EMPRESAS_DOMICILIOS")
public class EmpresaDomicilio extends CommonModel {
    @Column(name = "EMPRESA", nullable = false)
    private int empresa;

    @Column(name = "NOMBRE", nullable = false)
    private String nombre;

    @Column(name = "DOMICILIO_1", nullable = false)
    private String domicilio1;

    @Column(name = "NUMERO_EXTERIOR", nullable = false)
    private String numeroExterior;

    @Column(name = "NUMERO_INTERIOR")
    private String numeroInterior;

    @Column(name = "DOMICILIO_2", nullable = false)
    private String domicilio2;

    @Column(name = "DOMICILIO_3", nullable = false)
    private String domicilio3;

    @Column(name = "DOMICILIO_4")
    private String domicilio4;

    @Column(name = "CODIGO_POSTAL", nullable = false)
    private String codigoPostal;

    @Column(name = "ESTADO", nullable = false)
    private String estado;

    @Column(name = "PAIS", nullable = false)
    private String pais;

    @Column(name = "FOTO_FACHADA", nullable = false)
    private String fotoFachada;

    @Column(name = "MATRIZ", nullable = false)
    private boolean matriz;

    @Column(name = "TELEFONO_FIJO", nullable = false)
    private String telefonoFijo;

    @Column(name = "TELEFONO_MOVIL", nullable = false)
    private String telefonoMovil;

    @Column(name = "LATITUD", nullable = false)
    private String latitud;

    @Column(name = "LONGITUD", nullable = false)
    private String longitud;
}
